jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -157,6 +172,7 @@ public void recycle()
             _onFailure = null;
             _callbackFailure = null;
             _expects100Continue = false;
+            _complianceViolationListener = _complianceViolationListener.initialize();
```

Comment:
This is wrong, the field should be nulled out, not re-initialized.

A new listener could be created, for a connection that is possibly never used again and would idle timeout.

For HTTP/2 and HTTP/3, that may not recycle `HttpChannel`s, it is going to create a new listener that is never used.